09_json/n0npax/cmd/puppy-server/main_test.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	"bou.ke/monkey"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainNoArgs(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	args = []string{}
+	main()
+
+	expected := ""
+	actual := buf.String()
+	assert.Equal(t, expected, actual)
```

Comment:
Doesn't the program panic when you have no arguments? Maybe we have implemented differently.

Reply:
no it doesn't. It's just silent exit